### PR TITLE
Wired up setTimeout for natural user input (#17)

### PR DESCRIPTION
**Pull request by j3tchy**
Created `maskDelay` prop and `handleMaskDelay` method
Unit tests

### CODE REVIEW

File: src/DurationInputMask.js [hunk 1]
```diff
@@ -2,13 +2,18 @@ import React, { PureComponent } from 'react';
...
 
+// Polyfill
+import './polyfills/trimStart';
+
 matchAll.shim();
 
 const propTypes = {
   autoFocus: PropTypes.bool,
   component: PropTypes.node,
+  maskDelay: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (j3tchy) [reply src/DurationInputMask.js]: This doesn't work alongside maskDuration, it's replaced it.

File: src/DurationInputMask.test.js [hunk 2]
```diff
@@ -118,35 +119,50 @@ describe('Handles props', () => {
...
+
+    expect(wrapper.state('value')).toEqual();
   });
 
-  it('state not updated when props.maskOnChange=false & props.handleChange={fn}', () => {
-    const value = '2m';
+  it('props value is masked on setTimeout', (done) => {
+    const value = 61;
+    const delay = 250;
+    const delayInMilliseconds = delay / 1000;
```
Comment (nijk) [inline src/DurationInputMask.test.js]: Is this correct?

File: src/DurationInputMask.test.js [hunk 3]
```diff
@@ -118,35 +119,50 @@ describe('Handles props', () => {
...
+    const nextValue = '1m 1s';
+    const wrapper = shallow(
+      <DurationInputMask maskDelay={delay} value={value} />,
+    );
 
-    wrapper.simulate('change', event);
+    wrapper.simulate('blur');
 
-    expect(wrapper.state('value')).toEqual('');
+    expect(wrapper.state('value')).toEqual(nextValue);
```
Comment (nijk) [inline src/DurationInputMask.test.js]: I think we should be using `performance.now()` if asserting _when_ a state change happens.

**Comment by nijk** [comment]
@j3tchy I've fixed the conflict caused by #24 and added a `.gitattributes` file to provide a merge strategy for `.size-snapshot.json`.

File: src/DurationInputMask.test.js (hunk 3 above)
Comment (j3tchy) [reply src/DurationInputMask.test.js]: Use performance to check the state changed before the maskDelay

**Comment by j3tchy** [comment]
I've done a push the bugfix branch with the updated tests but there but a previous bug has resurfaced. It is not possible to clear the input field as `1s` always appears as the value when trying to clear the input field. This appears as the basic usage (with no maskDelay prop or set to 0). Will need to take a look at this before the PR is complete.

**Comment by nijk** [comment]
I think I know why that is happening, but the first task is to write a unit
test that catches the bug. Then you can set about fixing it

**Pull request merged**